Integrity sync test flaky behavior

## Description

During the Pre-Alpha 1 [system tests](https://github.com/wazuh/wazuh/issues/18016), it was found that the `test_cluster/test_integrity_sync/test_integrity_sync.py` system test for the `agentless_cluster` environment has a flaky behavior.

Sometimes a timeout error appears, and in other cases, a file that should be created is missing.

## Evidences
[integrity_sync_reports.zip](https://github.com/wazuh/wazuh-qa/files/12174467/integrity_sync_reports.zip)

## Issue Update

After running the test multiple times, it was found that the error is consistent and the message is always the same:

```
    def check_result(self):
        """Check if a TimeoutError occurred."""
        logger.debug(f'Checking results...')
        while not self._queue.empty():
            result = self._queue.get(block=True)
            for host, msg in result.items():
                if isinstance(msg, TimeoutError):
>                   raise msg
E                   TimeoutError: Did not found the expected callback in wazuh-master: .*Command received: b'cancel_task'.*

```

The error occurs on the _test_zip_size_limit_.

After reviewing the cluster.log of all components no error messages are found and after debugging the test it was found that the error occurs on the [_HostMonitor_](https://github.com/wazuh/wazuh-qa/blob/b262f030b65679d1b6ab580e672a3de6f28b48bf/tests/system/test_cluster/test_integrity_sync/test_integrity_sync.py#L277) class when executing the _run_ method.

And After debugging the [Run](https://github.com/wazuh/wazuh-qa/blob/b262f030b65679d1b6ab580e672a3de6f28b48bf/deps/wazuh_testing/wazuh_testing/tools/monitoring.py#L911) method it was found that the Timeout occurs because it can't find an available handler for the file monitor, the loop gets stuck there until a timeout occurs.

The next step would be to replicate the behavior of the test locally to ensure where is the error if it's an error on the test or the cluster.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -55,6 +55,7 @@ Release report: TBD
 - Fix a regex error in the FIM integration tests ([#3061](https://github.com/wazuh/wazuh-qa/issues/3061)) \- (Framework + Tests)
 - Fix an error in the cluster performance tests related to CSV parser ([#2999](https://github.com/wazuh/wazuh-qa/pull/2999)) \- (Framework + Tests)
 - Fix bug in the framework on migration tool ([#4027](https://github.com/wazuh/wazuh-qa/pull/4027)) \- (Framework)
+- Fix test cluster / integrity sync system test and configuration to avoid flaky behavior ([#4406](https://github.com/wazuh/wazuh-qa/pull/4406)) \- (Tests)
 
 ## [4.5.1] - TBD
 

--- a/tests/system/test_cluster/test_integrity_sync/data/cluster_json.yaml
+++ b/tests/system/test_cluster/test_integrity_sync/data/cluster_json.yaml
@@ -1,5 +1,5 @@
 ---
 timeout_receiving_file: 1
-max_zip_size: 52428800    # 50 MB
+max_zip_size: 104857600    # 100 MB
 min_zip_size: 15728640    # 15 MB
 compress_level: 0

--- a/tests/system/test_cluster/test_integrity_sync/test_integrity_sync.py
+++ b/tests/system/test_cluster/test_integrity_sync/test_integrity_sync.py
@@ -22,7 +22,7 @@ pytestmark = [pytest.mark.cluster, pytest.mark.agentless_cluster_env]
 test_hosts = ['wazuh-master', 'wazuh-worker1', 'wazuh-worker2']
 worker_hosts = test_hosts[1:]
 test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
-configuration = yaml.safe_load(open(os.path.join(test_data_path, 'cluster_json.yml')))
+configuration = yaml.safe_load(open(os.path.join(test_data_path, 'cluster_json.yaml')))
 messages_path = os.path.join(test_data_path, 'messages.yml')
 tmp_path = os.path.join(test_data_path, 'tmp')
 inventory_path = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
@@ -262,7 +262,7 @@ def test_zip_size_limit(clean_files, update_cluster_json):
     """
     too_big_size = configuration['max_zip_size'] + 1024
     big_size = configuration['min_zip_size'] - 1024
-    big_filenames = {file_prefix + str(i) for i in range(5)}
+    big_filenames = {file_prefix + str(i) for i in range(10)}
 
     # Create a tmp folder and all files inside in the master node.
     host_manager.run_command(test_hosts[0], f"mkdir {tmp_size_test_path}")

